server/container_create_linux.go

```diff
@@ -1119,6 +1116,36 @@ func addOCIBindMounts(ctx context.Context, ctr ctrfactory.Container, mountLabel,
 			options = append(options, "rprivate")
 		}
 
+		// Support for the Recursive Read-only (RRO) mount requires
+		// the mount to be read-only and the mount propagation to be
+		// set to private.
+		switch {
+		case m.Readonly && m.RecursiveReadOnly:
+			if !rroSupport {
+				return nil, nil, fmt.Errorf(
+					"recursive read-only mount support is not available for hostPath %q",
+					m.HostPath,
+				)
+			}
+			if m.Propagation != types.MountPropagation_PROPAGATION_PRIVATE {
```

Comment:
**To the reviewer:**

When deployed atop Kubernetes, we have the advantage of having the API request validation to tell us that required attributes are not set or have wrong values. However, when creating containers using the API, for example, using crictl, we don't have this added value, and as such, we perform this validation ourselves.

Reply:
I kinda feel we should do this validation in crictl to match kube doing it but I am open to double validating

Reply:
It depends on how you look at it. If you take CRI-O as an API server, then providing server-side validation, so to speak, would be more than fine, like the 412 HTTP status code in RESTful endpoints.